Corona-Warn-App/src/main/java/de/rki/coronawarnapp/datadonation/analytics/modules/keysubmission/AnalyticsKeySubmissionCollector.kt

```diff
@@ -15,98 +19,118 @@ import javax.inject.Inject
+    private val pcrStorage: AnalyticsPCRKeySubmissionStorage,
+    private val raStorage: AnalyticsRAKeySubmissionStorage,
     private val riskLevelStorage: RiskLevelStorage,
     private val riskLevelSettings: RiskLevelSettings
 ) {
 
-    fun reset() {
-        analyticsKeySubmissionStorage.clear()
+    fun reset(type: CoronaTest.Type) {
+        type.storage.clear()
     }
 
-    fun reportPositiveTestResultReceived() {
+    fun reportPositiveTestResultReceived(type: CoronaTest.Type) {
         if (disabled) return
-        analyticsKeySubmissionStorage.testResultReceivedAt.update { timeStamper.nowUTC.millis }
+        type.storage.testResultReceivedAt.update { timeStamper.nowUTC.millis }
     }
 
-    suspend fun reportTestRegistered() {
+    suspend fun reportTestRegistered(type: CoronaTest.Type) {
         if (disabled) return
+
         val testRegisteredAt = timeStamper.nowUTC
-        analyticsKeySubmissionStorage.testRegisteredAt.update { testRegisteredAt.millis }
+        type.storage.testRegisteredAt.update { testRegisteredAt.millis }
 
         val lastRiskResult = riskLevelStorage
             .latestAndLastSuccessfulEwRiskLevelResult
             .first()
             .tryLatestEwResultsWithDefaults()
             .lastCalculated
         val riskLevelAtRegistration = lastRiskResult.toMetadataRiskLevel()
-        analyticsKeySubmissionStorage.riskLevelAtTestRegistration.update {
+        type.storage.riskLevelAtTestRegistration.update {
             riskLevelAtRegistration.number
         }
 
         if (riskLevelAtRegistration == PpaData.PPARiskLevel.RISK_LEVEL_HIGH) {
-            riskLevelSettings.lastChangeToHighRiskLevelTimestamp?.let {
+            riskLevelSettings.ewLastChangeToHighRiskLevelTimestamp?.let {
                 val hours = Duration(
                     it,
                     testRegisteredAt
                 ).standardHours.toInt()
-                analyticsKeySubmissionStorage.hoursSinceHighRiskWarningAtTestRegistration.update {
+                type.storage.ewHoursSinceHighRiskWarningAtTestRegistration.update {
                     hours
                 }
             }
         }
 
-        analyticsKeySubmissionStorage.daysSinceMostRecentDateAtRiskLevelAtTestRegistration.update {
+        type.storage.ewDaysSinceMostRecentDateAtRiskLevelAtTestRegistration.update {
+            calculateDaysSinceMostRecentDateAtRiskLevelAtTestRegistration(
+                riskLevelSettings.ewMostRecentDateWithHighOrLowRiskLevel?.toLocalDateUtc(),
+                testRegisteredAt.toLocalDateUtc()
+            )
+        }
+
+        type.storage.ptDaysSinceMostRecentDateAtRiskLevelAtTestRegistration.update {
             calculateDaysSinceMostRecentDateAtRiskLevelAtTestRegistration(
-                riskLevelSettings.lastChangeCheckedRiskLevelTimestamp,
-                testRegisteredAt
+                riskLevelSettings.ptMostRecentDateWithHighOrLowRiskLevel,
```

Comment:
Is this missing `toLocalDateUtc()`?

Reply:
It is already a local date for PT. But I'm wondering if I even need this. I could just take it from RiskLevelStorage